### PR TITLE
Implement Power Logic, Pollution Logic and Processing Logic for MuTEs and many other things

### PR DESCRIPTION
**Pull request by BlueWeabo**
Features implemented:

- Power Logic
- Pollution Logic
- Processing Logic
- Wall-Sharable Parts
- Coke Oven testing out the Pollution logic and Processing Logic
- Coke Oven textured from @DragonStorm25 (aka Ant#3686)

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/machines/multiblock/logic/CokeOvenProcessingLogic.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        input.stackSize -= 1;
+        setDuration(NORMAL_RECIPE_TIME * timeMultiplier);
+        setOutputItems(output);
+
+        return originalStackSize > input.stackSize;
+    }
+
+    protected ItemStack findRecipe(ItemStack input) {
+        for (int oreId : OreDictionary.getOreIDs(input)) {
```
Comment (minecraft7771) [inline src/main/java/gregtech/common/tileentities/machines/multiblock/logic/CokeOvenProcessingLogic.java:42]: Wouldn't a switch be cleaner?

File: src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_Macerator.java
Comment (minecraft7771) [inline src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_Macerator.java:1]: Looking into the future: wouldn't it be better to manage the power object in a super class? We probably have more machines that handle power than not, which would result in much copy-pasted logic.

File: src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_CokeOven.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            tab.setDrawable(getGUITextureSet().getTitleTabDark()).setPos(0, -(titleHeight + TAB_PADDING * 2) + 1)
+                    .setSize(titleWidth + (TAB_PADDING + TITLE_PADDING) * 2, titleHeight + TAB_PADDING * 2 - 1);
+            text.setPos(TAB_PADDING + TITLE_PADDING, -titleHeight);
+        }
+        builder.widget(tab).widget(text);
+    }
+
+    @Override
+    public String getLocalName() {
+        return "Coke Oven";
```
Comment (minecraft7771) [inline src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_CokeOven.java:150]: Why no translation?

File: src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java [hunk 3]
```diff
@@ -235,142 +366,421 @@ protected IFluidTank[] getFluidTanks(byte aSide) {
...
+        }
+
+        if (hasThingsToDo()) {
+            markDirty();
+            runningTick(tick);
+        } else {
+            if (tick % TICKS_BETWEEN_RECIPE_CHECKS == 0 || hasWorkJustBeenEnabled() || hasInventoryBeenModified()) {
+                if (isAllowedToWork()) {
+                    wasEnabled = false;
+                    if (checkRecipe()) {
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java:462]: The regular multiblocks issue the processing start sound here, imo we should carry that over.

File: src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java [hunk 4]
```diff
@@ -235,142 +366,421 @@ protected IFluidTank[] getFluidTanks(byte aSide) {
...
 
-    @Override
-    public long getInputVoltage() {
-        return 1;
+    /**
+     * Runs only on server side
+     * 
+     * @param tick The current tick of the machine
+     */
+    protected void runningTick(long tick) {
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java:478]: I am not too sure where to put that in your structure, but runningTick should also play the processing sound of the multi. In regular multiblocks that is just done in the counterpart of this function, but on client side.

Comment (BlueWeabo) [reply src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java:478]: so a client side running Tick?

File: src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_CokeOven.java (hunk 2 above)
Comment (BlueWeabo) [reply src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_CokeOven.java:150]: Will at later (aka tomorrow)

File: src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java (hunk 4 above)
Comment (minecraft7771) [reply src/main/java/gregtech/api/multitileentity/machine/MultiTileBasicMachine.java:478]: Basically yes.

File: src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_Macerator.java
Comment (BlueWeabo) [reply src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_Macerator.java:1]: Do you mean as in PowerController super class? from where many multis could just automatically get a power logic?

Comment (minecraft7771) [reply src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_Macerator.java:1]: For example yes. The goal would just be to remove the need to save/load power to NBT for example.

Comment (BlueWeabo) [reply src/main/java/gregtech/common/tileentities/machines/multiblock/MultiBlock_Macerator.java:1]: will look into it

File: src/main/java/gregtech/common/tileentities/machines/multiblock/logic/CokeOvenProcessingLogic.java (hunk 1 above)
Comment (BlueWeabo) [reply src/main/java/gregtech/common/tileentities/machines/multiblock/logic/CokeOvenProcessingLogic.java:42]: Not possible with the way Java works.

**Review by minecraft7771 — APPROVED** [review]

**Pull request merged**